storage-xray-udp/src/main/java/zipkin2/storage/xray_udp/UDPMessageEncoder.java

```diff
@@ -55,8 +57,13 @@
         || span.kind() != Span.Kind.SERVER && span.kind() != Span.Kind.CONSUMER) {
       writer.name("type").value("subsegment");
       if (span.kind() != null) writer.name("namespace").value("remote");
+      writer.name("name").value(span.remoteServiceName() == null ? "" : span.remoteServiceName());
+    }else{
+      writer.name("name").value(span.localServiceName());
     }
-    writer.name("name").value(span.localServiceName());
+    // override with the user remote tag
```

Comment:
I don't think we should do it like this. we are setting the value of the xray field "namespace" to the value of a zipkin key "remote".

per amazon "Any calls to a remote service/broker should be namespace as ‘remote’. See ‘Optional Subsegment Fields’ under http://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-subsegments."

So, we should automatically default "namespace" to "remote" when it is a downstream PRODUCER or CLIENT span. If you look in the docs, the other supported choice is "aws". We don't support that in brave, yet, but we could.. in that case, I'd recommend using the zipkin key "xray.namespace" as a way to conditionally override.

make sense?

Reply:
`xray.namespace` sound good but all other variables are using something similar to `JSON path dot notation`. 

I will change it as you suggest.

Reply:
If I instrument a method (It might not be available at the time of this writing), what kind of span it will be? Since it is not connecting something else, I have some doubts on this.

Reply:
Im referring about a tag key ex span.tag("xray.namespace", "aws"). Reporter only sends complete spans.. how would we have a problem in routine use? For example there are only two values remote (we know) and aws (solved as soon as we write aws sdk instrumentation or integrate with it). Users will not do this manually in other words